client/wc-api/categories/operations.js

```diff
@@ -0,0 +1,54 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * WooCommerce dependencies
+ */
+import { stringifyQuery } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { isResourcePrefix, getResourceIdentifier, getResourceName } from '../utils';
+
+function read( resourceNames, fetch = apiFetch ) {
+	const filteredNames = resourceNames.filter( name => isResourcePrefix( name, 'category-query' ) );
+
+	return filteredNames.map( async resourceName => {
+		const query = getResourceIdentifier( resourceName );
+		const url = `/wp/v2/product_cat${ stringifyQuery( query ) }`;
```

Comment:
Can you use [the `wc/v3/product/categories` endpoint](https://woocommerce.github.io/woocommerce-rest-api-docs/#product-categories) here, or does the core taxonomy endpoint give you something else that you need?

Reply:
Good question @ryelle.  I initially used the core taxonomy endpoint because it allows us to override the collection param maximums so we can request more than 100 in the `per_page` param.

We can also do this with the WC API.  However, this doesn't remove the limitation and instead forces the `per_page` (currently set to `PHP_INT_MAX`) on every request to `wc/v3/product/categories`.

I implemented this in https://github.com/woocommerce/wc-admin/pull/1131/commits/32e5325f1bda9c620385fdd9882557896501a04d so you can take a look.  Which one do you think is a better approach?

Reply:
Are you trying to get all categories here? If you're using `apiFetch`, you can actually pass `per_page=-1` and it will work– behind the scenes [it fetches all items in batches of `per_page=100`,](https://github.com/WordPress/gutenberg/blob/master/packages/api-fetch/src/middlewares/fetch-all-middleware.js) and only returns to you when it's pulled all results. this is generally better on the server too, less chance of the API timing out.

Reply:
That's awesome!  I didn't know we had this.  Thanks @ryelle .

I prematurely merged this before I saw your comment, but opened an issue to handle this - https://github.com/woocommerce/wc-admin/issues/1148